pkg/cloudprovider/aws/apis/v1alpha1/provider_defaults.go

```diff
@@ -75,3 +77,24 @@ func (c *Constraints) defaultSecurityGroups(clusterName string) {
 	}
 	c.SecurityGroupSelector = map[string]string{fmt.Sprintf(ClusterDiscoveryTagKeyFormat, clusterName): "*"}
 }
+
+const (
+	DefaultMetadataOptionsHTTPEndpoint            = ec2.LaunchTemplateInstanceMetadataEndpointStateEnabled
+	DefaultMetadataOptionsHTTPPutResponseHopLimit = 2
+	DefaultMetadataOptionsHTTPTokens              = ec2.LaunchTemplateHttpTokensStateOptional
+)
+
+func (c *Constraints) defaultMetadataOptions() {
+	if c.MetadataOptions == nil {
+		c.MetadataOptions = &MetadataOptions{}
+	}
+	if c.MetadataOptions.HTTPEndpoint == "" {
+		c.MetadataOptions.HTTPEndpoint = DefaultMetadataOptionsHTTPEndpoint
+	}
+	if c.MetadataOptions.HTTPPutResponseHopLimit == 0 {
```

Comment:
Maybe a dumb question. Is it possible that a user would want to configure the hop to 0? Does this mean anything? If so, we'd want to make this a pointer so it can be left nil.

Reply:
The AWS API only allows values from 1 to 64. I don't think there is a valid case for 0.